### PR TITLE
[FIRRTL] Verify RWProbeOp target has layer requirements.

### PR DESCRIPTION
**Pull request by dtzSiFive**
RWProbe conservatively means a write to the target, so check that the target is indeed writeable from where the rwprobe is.

### CODE REVIEW

File: test/Dialect/FIRRTL/errors.mlir [hunk 1]
```diff
@@ -2072,6 +2072,20 @@ firrtl.circuit "RWProbeUseDef" {
...
+  firrtl.layer @A bind { }
+  firrtl.module @RWProbeLayerRequirements(in %cond : !firrtl.uint<1>) {
+    // expected-note @below {{target is missing layer requirements: @A}}
+    %w = firrtl.wire sym @x : !firrtl.uint<1>
+    firrtl.layerblock @A {
+      // expected-error @below {{target has insufficient layer requirements}}
+      %rw = firrtl.ref.rwprobe <@RWProbeLayerRequirements::@x> : !firrtl.rwprobe<uint<1>>
+    }
+  }
+}
```
Comment (seldridge) [inline test/Dialect/FIRRTL/errors.mlir:2085]: Are there any checks of _legally_ doing this, i.e., what should be allowed here? If there are not, can one be added?

**Review by seldridge — APPROVED** [review]
LGTM

File: test/Dialect/FIRRTL/errors.mlir (hunk 1 above)
Comment (dtzSiFive) [reply test/Dialect/FIRRTL/errors.mlir:2085]: Good call, added one.  Thanks!

**Comment by dtzSiFive** [comment]
Thanks!

**Pull request merged**